python/BioSimSpace/Align/_align.py

```diff
@@ -1297,13 +1763,140 @@ def flexAlign(
+        "rmsd_flex_align" function. If None is passed, then BioSimSpace
+        will attempt to find fkcombu by searching your PATH.
+
+    property_map0 : dict
+        A dictionary that maps "properties" in molecule0 to their user
+        defined values. This allows the user to refer to properties
+        with their own naming scheme, e.g. { "charge" : "my-charge" }
+
+    property_map1 : dict
+        A dictionary that maps "properties" in molecule1 to their user
+        defined values.
+
+    Returns
+    -------
+
+    molecule : :class:`Molecule <BioSimSpace._SireWrappers.Molecule>`
+        The aligned molecule.
+    """
+
+    if not isinstance(molecule0, _Molecule):
+        raise TypeError(
+            "'molecule0' must be of type 'BioSimSpace._SireWrappers.Molecule'"
+        )
+
+    if not isinstance(molecule1, _Molecule):
+        raise TypeError(
+            "'molecule1' must be of type 'BioSimSpace._SireWrappers.Molecule'"
+        )
+    if roi is None:
+        raise ValueError("No region of interest (ROI) has been provided.")
+    else:
+        if not isinstance(roi, list):
+            raise TypeError("'roi' must be of type 'list'.")
+
+    if align_function not in ["rmsd", "rmsd_flex_align"]:
+        raise ValueError(
+            "Invalid alignment function. Available options are: 'rmsd', 'rmsd_flex_align'"
+        )
+
+    for roi_idx in roi:
+        res0 = molecule0.getResidues()[roi_idx]
+        res1 = molecule1.getResidues()[roi_idx]
+
+        res0_idx = [a.index() for a in res0.getAtoms()]
+        res1_idx = [a.index() for a in res1.getAtoms()]
+
+        # Extract the residues of interest from the molecules
+        res0_extracted = molecule0.extract(res0_idx)
+        res1_extracted = molecule1.extract(res1_idx)
+
+        if align_function == "rmsd":
+            res0_aligned = rmsdAlign(molecule0=res0_extracted, molecule1=res1_extracted)
+
+        elif align_function == "rmsd_flex_align":
+            res0_aligned = flexAlign(
+                molecule0=res0_extracted,
+                molecule1=res1_extracted,
+                fkcombu_exe=fkcombu_exe,
+            )
+
+        # Now update molecule0 with the aligned residue coordinates
+        mol0 = molecule0._getSireObject()
+        res0_aligned_coords = res0_aligned._getSireObject().property("coordinates")
+
+        # A list to store the updated coordinates for molecule0
+        mol0_coords = []
+        for i, res in enumerate(mol0.residues()):
+            if i == roi_idx:
+                mol0_coords.append(res0_aligned_coords)
+            else:
+                mol0_coords.append(res.property("coordinates"))
+
+        # Flatten the list
+        mol0_coords = [item for sublist in mol0_coords for item in sublist]
+
+        # Create a cursor for updating the coordinates property
+        c = mol0.cursor()
+        for i, atom in enumerate(c.atoms()):
+            atom["coordinates"] = mol0_coords[i]
```

Comment:
The property might not be called `coordinates`, so we need to use the passed map to get the actual name, i.e. `property_map0.get("coordinates", "coordinates")`.

Reply:
Thanks, I added code to get the property name from the map if provided.